test/cpp/end2end/grpc_authz_end2end_test.cc

```diff
@@ -533,38 +529,36 @@ TEST_F(GrpcAuthzEnd2EndTest, StaticInitWithAuditLoggingOnDeny) {
   grpc::Status status;
 
   ClientContext context1;
+  // Matches the allow rule.
   context1.AddMetadata("key-foo", "foo");
   status = SendRpc(channel, &context1, &resp);
   EXPECT_TRUE(status.ok());
-  EXPECT_EQ(log_count_, 0);
+  EXPECT_EQ(audit_logs_.size(), 0);
 
   ClientContext context2;
+  // Does not match the allow rule or deny rule.
   context2.AddMetadata("key-foo", "bar");
-  ::testing::internal::CaptureStdout();
   status = SendRpc(channel, &context2, &resp);
-  CompareAuditLogEntry(
-      ::testing::internal::GetCapturedStdout(),
-      "{\"authorized\":false,\"matched_rule\":\"\",\"policy_name\":\"authz\","
-      "\"principal\":\"spiffe://foo.com/bar/baz\",\"rpc_method\":\"/"
-      "grpc.testing.EchoTestService/Echo\"}");
   EXPECT_EQ(status.error_code(), grpc::StatusCode::PERMISSION_DENIED);
   EXPECT_EQ(status.error_message(), "Unauthorized RPC request rejected.");
-  EXPECT_EQ(log_count_, 1);
+  EXPECT_EQ(audit_logs_.size(), 1);
 
   ClientContext context3;
-  context3.AddMetadata("key-foo", "foo");
+  // Matches the deny rule.
   context3.AddMetadata("key-bar", "bar");
-  ::testing::internal::CaptureStdout();
   status = SendRpc(channel, &context3, &resp);
-  CompareAuditLogEntry(
-      ::testing::internal::GetCapturedStdout(),
-      "{\"authorized\":false,\"matched_rule\":\"deny_bar\",\"policy_name\":"
-      "\"authz\","
-      "\"principal\":\"spiffe://foo.com/bar/baz\",\"rpc_method\":\"/"
-      "grpc.testing.EchoTestService/Echo\"}");
   EXPECT_EQ(status.error_code(), grpc::StatusCode::PERMISSION_DENIED);
   EXPECT_EQ(status.error_message(), "Unauthorized RPC request rejected.");
-  EXPECT_EQ(log_count_, 2);
+  EXPECT_EQ(audit_logs_.size(), 2);
+  EXPECT_THAT(
+      audit_logs_,
```

Comment:
I think it would be a little clearer to check the contents of the first log after its associated request, then the last log after the last request rather than all of them at the end

Reply:
Probably because I wrote this, either way seems equally clear to me. But I changed it so that more people are happy :)